packages/hd/tests/keystore.test.ts

```diff
@@ -44,6 +44,34 @@ test("load and check password, loads private key", (t) => {
   t.is(extendedPrivateKey.chainCode, fixture.chainCode);
 });
 
+// test vector:
+// https://github.com/ethereum/wiki/wiki/Web3-Secret-Storage-Definition#test-vectors
+test("load test vector keystore", (t) => {
+  const json = {
+    crypto: {
+      cipher: "aes-128-ctr",
+      cipherparams: {
+        iv: "83dbcc02d8ccb40e466191a123791e0e",
+      },
+      ciphertext:
+        "d172bf743a674da9cdad04534d56926ef8358534d458fffccd4e6ad2fbde479c",
+      kdf: "scrypt",
+      kdfparams: {
+        dklen: 32,
+        n: 262144,
+        p: 8,
+        r: 1,
+        salt: "ab0c7876052600dd703518d6fc3fe8984592145b591fc8fb5c6d43190334ba19",
+      },
+      mac: "2103ac29920d71da29f15d75b4a16dbe95cfd7ff8faea1056c33131d846e3097",
+    },
+    id: "3198bc9c-6672-5ab3-d995-4942343ae5b6",
+    version: 3,
+  };
+  const keystore = Keystore.fromJson(JSON.stringify(json));
+  t.true(keystore.checkPassword("testpassword"));
```

Comment:
we can decrypt the json here to check if the decrypted secret is same with original secret

Reply:
sure, added assertion for `decrypt` and `derivekey`